cmd/kubeadm/app/phases/kubelet/flags.go

```diff
@@ -96,13 +96,20 @@ func buildKubeletArgMap(opts kubeletFlagsOpts) map[string]string {
 		kubeletFlags["container-runtime-endpoint"] = opts.nodeRegOpts.CRISocket
 	}
 
-	if opts.registerTaintsUsingFlags && opts.nodeRegOpts.Taints != nil && len(opts.nodeRegOpts.Taints) > 0 {
-		taintStrs := []string{}
+	// Taint the Node using the kubelet flag --register-with-taints
+	taintStrs := []string{}
+	if opts.addControlPlaneTaint {
```

Comment:
We already have this taint in the list https://github.com/kubernetes/kubernetes/blob/bd4c9a2263a3828081a850907326ebdf14cb242a/cmd/kubeadm/app/apis/kubeadm/v1beta2/types.go#L206

Reply:
yes, but my PR also removes it from being automatically added to NodeRegistrationOptions.
i guess this is one way of handling this.